packages/devtools-components/src/tree.js

```diff
@@ -138,7 +140,10 @@ const TreeNode = createFactory(createClass({
       renderItem(item, depth, focused, arrow, expanded)
     );
   }
-}));
+}
+
+const ArrowExpanderFactory = createFactory(ArrowExpander);
```

Comment:
I didn't know that createFactory() is set to be deprecated (there is a TODO in the source).

Using new ArrowExpander() fails because you need a ReactClass and not a standard Class. The solution was to just replace createFactory with createElement calls, which is what the React folk recommend.

I have done this with all the crateFactory calls so that we don't need to change them in the future.